src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Location/ShowLocationHistory/ShowLocationHistory.xaml.cs

```diff
@@ -0,0 +1,160 @@
+            // Create the location data source.
+            _locationDataSource = new FakeLocationDataSource();
+            // Use this instead if you want real location: _locationDataSource = new SystemLocationDataSource();
+
+            try
+            {
+                // Start the data source.
+                await _locationDataSource.StartAsync();
+
+                if (_locationDataSource.IsStarted)
+                {
+                    // Set the location display data source and enable location display.
+                    MyMapView.LocationDisplay.DataSource = _locationDataSource;
+                    MyMapView.LocationDisplay.IsEnabled = true;
+                    MyMapView.LocationDisplay.AutoPanMode = LocationDisplayAutoPanMode.Recenter;
+                    MyMapView.LocationDisplay.InitialZoomScale = 10000;
+
+                    // Enable the button to start location tracking.
+                    LocationTrackingButton.IsEnabled = true;
+                }
+                else
+                {
+                    ShowMessage("There was a problem enabling location", "Error");
+                }
+            }
+            catch (Exception e)
+            {
+                System.Diagnostics.Debug.WriteLine(e);
+                ShowMessage("There was a problem enabling location", "Error");
+            }
+        }
+
+        private void ToggleLocationTracking()
+        {
+            // Toggle location tracking.
+            _isTrackingEnabled = !_isTrackingEnabled;
+
+            // Apply new configuration.
+            if (_isTrackingEnabled)
+            {
+                // Configure new symbology first.
+                _locationDataSource.LocationChanged += LocationDataSourceOnLocationChanged;
+
+                // Update the UI.
+                LocationTrackingButton.Content = "Stop tracking";
+            }
+            else
+            {
+                // Stop updating.
+                _locationDataSource.LocationChanged -= LocationDataSourceOnLocationChanged;
+
+                // Update the UI.
+                LocationTrackingButton.Content = "Start tracking";
+            }
+        }
+
+        private void LocationDataSourceOnLocationChanged(object sender, Location e)
+        {
+            // Remove the old line.
+            _locationHistoryLineOverlay.Graphics.Clear();
+
+            // Add any previous position to the history.
+            if (_lastPosition != null)
+            {
+                _polylineBuilder.AddPoint(_lastPosition);
+                _locationHistoryOverlay.Graphics.Add(new Graphic(_lastPosition));
+            }
+
+            // Store the current position.
+            _lastPosition = e.Position;
+
+            // Add the updated line.
+            _locationHistoryLineOverlay.Graphics.Add(new Graphic(_polylineBuilder.ToGeometry()));
+        }
+
+        private void LocationTrackingButton_OnClick(object sender, RoutedEventArgs e) => ToggleLocationTracking();
```

Comment:
I am unsure why this is written using a separate `ToggleLocationTracking()` method. Why not put that code in 'LocationTrackingButton_OnClick' . I am sure there is a reason, and I am genuinely asking.

Reply:
I did it to keep the platform-specific UI code separate from the core sample code as much as possible. It also allows me to give that method a meaningful name that doesn't need to follow the event handler naming rules.